drivers/gpio/gpio_gecko.c

```diff
@@ -26,11 +26,7 @@
 #if DT_NODE_HAS_PROP(id, peripheral_id)
 #define GET_GECKO_GPIO_INDEX(id) DT_INST_PROP(id, peripheral_id)
 #else
-#if defined(CONFIG_SOC_SERIES_EFR32BG22) || \
-	defined(CONFIG_SOC_SERIES_EFR32BG27) || \
-	defined(CONFIG_SOC_SERIES_EFR32MG21) || \
-	defined(CONFIG_SOC_SERIES_EFR32MG24) || \
-	defined(CONFIG_SOC_SERIES_EFR32ZG23)
+#if defined(CONFIG_SOC_FAMILY_SILABS_S2)
```

Comment:
This change and the relocation of `soc.h` are orthogonal to the support for xg29. Could you place them in a separate commit(s)?

Reply:
Done.